x-pack/test/performance/tests/playwright/ecommerce_dashboard.ts

```diff
@@ -5,52 +5,68 @@
  * 2.0.
  */
 import Url from 'url';
+import { Page } from 'playwright';
 import { FtrProviderContext } from '../../ftr_provider_context';
 
 export default function ecommerceDashboard({ getService }: FtrProviderContext) {
   describe('ecommerce_dashboard', () => {
-    const config = getService('config');
-    const performance = getService('performance');
-    const logger = getService('log');
+    it('ecommerce_dashboard', async () => {
+      const config = getService('config');
+      const performance = getService('performance');
+      const logger = getService('log');
 
-    const { step } = performance.makePage('ecommerce_dashboard');
+      await performance.runUserJourney(
+        'ecommerce_dashboard',
+        [
+          {
+            name: 'Go to Sample Data Page',
+            handler: async ({ page }: { page: Page }) => {
+              const kibanaUrl = Url.format({
+                protocol: config.get('servers.kibana.protocol'),
+                hostname: config.get('servers.kibana.hostname'),
+                port: config.get('servers.kibana.port'),
+              });
 
-    step('Go to Sample Data Page', async ({ page }) => {
-      const kibanaUrl = Url.format({
-        protocol: config.get('servers.kibana.protocol'),
-        hostname: config.get('servers.kibana.hostname'),
-        port: config.get('servers.kibana.port'),
-      });
+              await page.goto(`${kibanaUrl}/app/home#/tutorial_directory/sampleData`);
+              await page.waitForSelector('text="More ways to add data"');
+            },
+          },
+          {
+            name: 'Add Ecommerce Sample Data',
+            handler: async ({ page }: { page: Page }) => {
+              const removeButton = page.locator('[data-test-subj=removeSampleDataSetecommerce]');
```

Comment:
Can we use the `PageObjects` pattern here?

Reply:
I'm not sure if you mean creating a class represents a certain page like [here](https://www.selenium.dev/documentation/test_practices/encouraged/page_object_models/) and like in our functional tests. If that's what you mean we can discuss advantages/disadvantages single user journeys but that will change the scope of this PR pretty radically.